### PR TITLE
Open login overlay if user isn't signed in after intro sequence

### LINKED ISSUE #6030: Login prompt should automatically appear if user isn’t logged in.
Upon entering the game, login prompt should appear once clicking the osu! logo if user isn’t signed in. Or, it appears when intro sequence ends, matching stable behaviour.

### PR DESCRIPTION
**Pull request by LeNitrous**
Matches osu-stable behavior. Resolves #6030. Supersedes #6037.

### CODE REVIEW

File: osu.Game/Screens/Menu/MainMenu.cs [hunk 1]
```diff
@@ -128,6 +135,9 @@ public override void OnEntering(IScreen last)
                     track.Seek(metadata.PreviewTime != -1 ? metadata.PreviewTime : 0.4f * track.Length);
                     track.Start();
                 }
+
+                if (api?.State == APIState.Offline)
```
Comment (peppy) [inline osu.Game/Screens/Menu/MainMenu.cs]: which is this inside a track-related conditional?

Comment (LeNitrous) [reply osu.Game/Screens/Menu/MainMenu.cs]: didn't realize it was checking the track as well, I'll make a separate clause where it only checks if the last screen was an `IntroScreen`

File: osu.Game/Screens/Menu/MainMenu.cs [hunk 2]
```diff
@@ -130,6 +137,9 @@ public override void OnEntering(IScreen last)
                 }
             }
 
+            if (last is IntroScreen && api?.State == APIState.Offline)
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Menu/MainMenu.cs]: never null
```suggestion
            if (last is IntroScreen && api.State == APIState.Offline)
```

**Comment by peppy** [comment]
You probably want `api.IsLoggedIn` instead (should be safer). Also, I don't think you need the `last is IntroScreen` check – have you investigated why that is there?

**Comment by LeNitrous** [comment]
Oh right. `MainMenu` just suspends into the background when transitioning to Song Select or Editor.

**Comment by peppy** [comment]
This looks visually jarring. THe login overlay should not be shown without this toolbar present.

**Comment by peppy** [comment]
I've fixed some remaining issues up. Seems okay to me now.

**Review by peppy — DISMISSED** [review]

**Review by peppy — DISMISSED** [review]

**Pull request merged**